### ISSUE TITLE
[13] Multi-Task Deep Neural Networks for Natural Language Understanding (MT-DNN)

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#40]
[paper](https://arxiv.org/pdf/1901.11504.pdf)
[code](https://github.com/namisan/mt-dnn)
[youtube - TMax AI](https://www.youtube.com/watch?v=rG8A1VFSENM)
[blog - 한국어 리뷰](https://y-rok.github.io/nlp/2019/05/20/mt-dnn.html)

## MT-DNN
여러 개의 task 를 multi-task learning 시켜서 BERT 보다 뛰어난 성능 만들기.

![image](https://user-images.githubusercontent.com/16400591/137855673-d31759dc-a624-4439-b6bb-7e002953f020.png)
#### Classification
![image](https://user-images.githubusercontent.com/16400591/137879757-323ef1ce-cfb8-41e0-b668-f2bb4e9742a5.png)
CLS token 에 Linear 통과 후 softmax

#### Text Similarity Output
![image](https://user-images.githubusercontent.com/16400591/137879850-7085c83b-97ee-4c14-8f6e-8b858aaba00f.png)

#### Pairwise Text Classification
[SAN (Stochastic Answer Network)](https://arxiv.org/abs/1804.07888) 을 이용해서 추가적으로 task 풀어냄
CLS token을 쓰지 않고, token embedding vector 를 활용해서 SAN을 태우게 됨.
k번 prediction을 통해 보다 정확한 prediction 수행. (최종 성능은 CLS token 쓸 때보다 1% 정도 올라감)
![image](https://user-images.githubusercontent.com/16400591/137879877-357f26e3-a9c9-4be3-909f-3ff502fb104c.png)
각 문장의 vector, 거리, similarity 를 concat 해서 weight 태움

#### Relevance Ranking Output
질문 문장이 주어지면, 지문 중 정답이 있는 문장을 ranking을 통해 찾아냄!
Question과 문장 Pair를 Input으로 넣어 생성한 [CLS] Token에 Sigmoid를 취함
![image](https://user-images.githubusercontent.com/16400591/137887638-2a8c75f4-ea71-44bc-bdfc-b3cc49b140d7.png)

## Algorithm
![image](https://user-images.githubusercontent.com/16400591/137888168-7517267b-c0be-497f-8074-c2ccc592a101.png)
배치는 하나의 task로만 이루어진 상태.
~일반적인 MTL setting과는 다른 듯?~

## Results
![image](https://user-images.githubusercontent.com/16400591/137887240-595f67a6-0890-4a4e-b437-e404b092c6e8.png)
